borsh/src/de/mod.rs

```diff
@@ -294,6 +297,53 @@ impl BorshDeserialize for String {
     }
 }
 
+#[cfg(feature = "bigdecimal")]
+impl BorshDeserialize for bigdecimal::BigDecimal {
+    #[inline]
+    fn deserialize(buf: &mut &[u8]) -> Result<Self> {
+        let digits = bigdecimal::num_bigint::BigInt::deserialize(buf)?;
+        let scale = i64::deserialize(buf)?;
+        Ok(bigdecimal::BigDecimal::new(digits, scale))
+    }
+}
+
+#[cfg(feature = "bigdecimal")]
+impl BorshDeserialize for bigdecimal::num_bigint::BigInt {
```

Comment:
I just wonder if this implementation should come from a separate feature and implement it directly from that crate. Just thinking of an edge case where someone might want to have `num_bigint` impls without having to compile `bigdecimal`.

Even if we wanted to do this or if someone has an issue, we should be able to do later without it being breaking change, so I think fine as is